### PR TITLE
feat(@angular/cli): Allow ability to set budget sizes for your bundles

### LINKED ISSUE #7139: Allow setting of budget of bundle sizes
### Bug Report or Feature Request (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [x] feature request
```

### Desired functionality.
Allow users to set size budgets for bundles created.
Once they're defined production builds will fail if the specified allowed sizes are exceeded.

### PR DESCRIPTION
**Pull request by Brocco**
Closes #7139

### CODE REVIEW

File: packages/@angular/cli/lib/config/schema.json [hunk 1]
```diff
@@ -192,6 +192,22 @@
             "description": "Name and corresponding file for environment config.",
             "type": "object",
             "additionalProperties": true
+          },
+          "budgets": {
+            "description": "Defines the budget allotments for bundles.",
+            "type": "array",
+            "items": {"type": "object",
```
Comment (hansl) [inline packages/@angular/cli/lib/config/schema.json]: Split in 2 lines.

File: packages/@angular/cli/lib/config/schema.json [hunk 2]
```diff
@@ -192,6 +192,22 @@
...
+              "properties": {
+                "name": {
+                  "description": "Name of the bundle.",
+                  "type": "string"
+                },
+                "budget": {
+                  "description": "Threshold to measure against. (in kb)",
```
Comment (hansl) [inline packages/@angular/cli/lib/config/schema.json]: I'd rather have a string or number, and string can contain `g` `M` or `k` or whatever.

File: packages/@angular/cli/lib/config/schema.json [hunk 3]
```diff
@@ -192,6 +192,22 @@
             "description": "Name and corresponding file for environment config.",
             "type": "object",
             "additionalProperties": true
+          },
+          "budgets": {
```
Comment (hansl) [inline packages/@angular/cli/lib/config/schema.json]: With this scheme, how do I specify the error/warning levels? How do I specify my budget for initial vs total?

**Comment by Brocco** [comment]
This PR is the original version of the already merged #8939, closing.

**Pull request closed without merging**